starport/templates/ibc/oracle.go

```diff
@@ -0,0 +1,370 @@
+	return func(r *genny.Runner) error {
+		path := fmt.Sprintf("x/%s/client/cli/query.go", opts.ModuleName)
+		f, err := r.Disk.Find(path)
+		if err != nil {
+			return err
+		}
+		template := `%[1]v
+
+	cmd.AddCommand(Cmd%[2]vResult())
+	cmd.AddCommand(CmdLast%[2]vID())
+`
+		replacement := fmt.Sprintf(template, Placeholder, opts.OracleName.UpperCamel)
+		content := replacer.Replace(f.String(), Placeholder, replacement)
+		newFile := genny.NewFileS(path, content)
+		return r.File(newFile)
+	}
+}
+
+func clientCliTxOracleModify(replacer placeholder.Replacer, opts *OracleOptions) genny.RunFn {
+	return func(r *genny.Runner) error {
+		path := fmt.Sprintf("x/%s/client/cli/tx.go", opts.ModuleName)
+		f, err := r.Disk.Find(path)
+		if err != nil {
+			return err
+		}
+		template := `%[1]v
+	cmd.AddCommand(CmdRequest%[2]vData())
+`
+		replacement := fmt.Sprintf(template, Placeholder, opts.OracleName.UpperCamel)
+		content := replacer.Replace(f.String(), Placeholder, replacement)
+		newFile := genny.NewFileS(path, content)
+		return r.File(newFile)
+	}
+}
+
+func codecOracleModify(replacer placeholder.Replacer, opts *OracleOptions) genny.RunFn {
+	return func(r *genny.Runner) error {
+		path := fmt.Sprintf("x/%s/types/codec.go", opts.ModuleName)
+		f, err := r.Disk.Find(path)
+		if err != nil {
+			return err
+		}
+
+		// Set import if not set yet
+		replacement := `sdk "github.com/cosmos/cosmos-sdk/types"`
+		content := replacer.ReplaceOnce(f.String(), Placeholder, replacement)
+
+		// Register the module packet
+		templateRegistry := `%[1]v
+cdc.RegisterConcrete(&Msg%[3]vData{}, "%[2]v/%[3]vData", nil)
+`
+		replacementRegistry := fmt.Sprintf(templateRegistry, Placeholder2, opts.ModuleName, opts.OracleName.UpperCamel)
+		content = replacer.Replace(content, Placeholder2, replacementRegistry)
+
+		// Register the module packet interface
+		templateInterface := `%[1]v
+registry.RegisterImplementations((*sdk.Msg)(nil),
+	&Msg%[2]vData{},
+)`
+		replacementInterface := fmt.Sprintf(templateInterface, Placeholder3, opts.OracleName.UpperCamel)
+		content = replacer.Replace(content, Placeholder3, replacementInterface)
+
+		newFile := genny.NewFileS(path, content)
+		return r.File(newFile)
+	}
+}
+
+func packetHandlerOracleModify(replacer placeholder.Replacer, opts *OracleOptions) genny.RunFn {
+	return func(r *genny.Runner) error {
+		path := fmt.Sprintf("x/%s/oracle.go", opts.ModuleName)
+		f, err := r.Disk.Find(path)
+		if err != nil {
+			return err
+		}
+
+		// Register the module packet
+		templateRecv := `%[1]v
+	var %[2]vResult types.%[3]vResult
+	if err := obi.Decode(modulePacketData.Result, &%[2]vResult); err == nil {
+		am.keeper.Set%[3]vResult(ctx, types.OracleRequestID(modulePacketData.RequestID), %[2]vResult)
+		ack = channeltypes.NewResultAcknowledgement(
+			types.ModuleCdc.MustMarshalJSON(
+				packet.NewOracleRequestPacketAcknowledgement(modulePacketData.RequestID),
+			),
+		)
+
+		// TODO: %[3]v oracle data reception logic
```

Comment:
Thanks for adding this! After having though a bit more about it I may think we don't need it at the end.

By default the last result is set in the store with `SetFooResult`, and so maybe instead to telling developers to put their custom logic here, a better practice is that their custom logic that would interpret crypto prices would use `GetFooResult` and would be placed somewhere else, and we precise it in the doc. And so the scaffolded query is purely designed to just get the result.

This is open to discussion

Reply:
I like the placeholder for the users to know what's needed to implement if some packages coming. And they can use in real-time or from the kv database